### PR TITLE
Replace AppKernel version constants by prestashop versions

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | This PR aims to replaces all AppKernel references by a new Prestashop that contains the version. There is some services inside the prestashop bundle that contains AppKernel references and will throw errors if the bundle is used without AppKernel. AppKernel constants will not depend on this constants instead of defining it.
| Type?             | improvement
| Category?         | IN
| BC breaks?        |  no
| Deprecations?     | no

### CODE REVIEW

File: app/AppKernel.php [hunk 1]
```diff
@@ -54,7 +56,7 @@ public function registerBundles()
             // PrestaShop Core bundle
             new PrestaShopBundle\PrestaShopBundle(),
             // PrestaShop Translation parser
-            new PrestaShop\TranslationToolsBundle\TranslationToolsBundle(),
+            new TranslationToolsBundle(),
```
Comment (PululuK) [inline app/AppKernel.php:59]: seem not related to the PR ?

**Review by PululuK — DISMISSED** [review]

File: app/AppKernel.php (hunk 1 above)
Comment (FabienPapet) [reply app/AppKernel.php:59]: I could not leave it as it is. Because of the Prestashop class name that create conflicts with this namespace.

**Review by nicosomb — DISMISSED** [review]

**Review by matks — COMMENTED** [review]
This sounds a good idea. Just to be sure 😄 I'd like some other "old people" like me opinion about such a change: maybe @kpodemski ? @Matt75 ? @atomiix ?

And if we merge this we'll need to update "how to modify version number" part of https://www.prestashop-project.org/maintainers-guide/releasing-prestashop/preliminary-tasks/

File: src/Core/Prestashop.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace PrestaShop\PrestaShop\Core;
+
+class Prestashop
```
Comment (jolelievre) [inline src/Core/Prestashop.php]: Can you please add a quick description of this class please? Even if it's just about describing that it's just meant to be a "const container"

By the way you could add `private __construct()` method to emphasize (and even block) the fact that this lass should not be instanciated

**Review by jolelievre — COMMENTED** [review]
Thanks @FabienPapet 

Just one feedback, and I think this change will have some impact on the TranslationTool app because it gets the PrestaShop target version by parsing the AppKernel const So it will need to be able to get it from the kernel or the new class in the future I'm gonna create an issue on the repo

File: src/Core/Prestashop.php (hunk 2 above)
Comment (PululuK) [reply src/Core/Prestashop.php]: @jolelievre so, It can be abstract, right ?

Comment (jolelievre) [reply src/Core/Prestashop.php]: Related issue created https://github.com/PrestaShopCorp/TranslationTool/issues/120

Comment (PululuK) [reply src/Core/Prestashop.php]: image

**Comment by kpodemski** [comment]
Maybe it is a nitpick, but I'm not sure about the naming.

`PrestaShop/PrestaShop/Core/Prestashop`

Why not `PrestaShop/PrestaShop/Core/Version` if it's all about the version?

`PrestaShop/PrestaShop/Core/Version::CURRENT`
`PrestaShop/PrestaShop/Core/Version::MAJOR`, etc.

**Comment by FabienPapet** [comment]
I renamed the `Prestashop` class to `Version`, added a class description and a private constructor.

**Comment by FabienPapet** [comment]
Documentation PR https://github.com/PrestaShop/open-source/pull/126/files

**Review by matks — DISMISSED** [review]

**Review by NeOMakinG — DISMISSED** [review]

**Comment by kpodemski** [comment]
There's no "How to test" section. I guess automatic tests should be enough. Right?

@PrestaShop/qa-team

**Review by jolelievre — DISMISSED** [review]

**Comment by jolelievre** [comment]
@kpodemski I agree this one probably doesn't need functional QA, I just ran the auto campaigns on this PR Here is the run https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/3241666126

We can validate with @PrestaShop/qa-team if the full campaign is green I guess it should be enough?

**Comment by FabienPapet** [comment]
@PrestaShop/qa-team friendly reminder

**Comment by Progi1984** [comment]
@FabienPapet Could you give a scenario test which allows to test each change in your code ?

**Comment by FabienPapet** [comment]
I think there's no need for a test suite here. Running PS in dev env should display the right version of PS in Symfony debug toolbar.

**Comment by FabienPapet** [comment]
I've added a test to prevent the changes and keep versions in sync

**Review by nicosomb — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Review by khouloudbelguith — APPROVED** [review]
Hi @FabienPapet,

Automatic tests OK :heavy_check_mark: 
I checked the Prestashop version => OK

https://user-images.githubusercontent.com/16067358/198278590-be23edd3-9f15-4440-a3b4-141458fea20b.mp4

Thanks!

**Comment by jolelievre** [comment]
Thanks @FabienPapet and @khouloudbelguith
